test/helpers/integration.rb

```diff
@@ -167,7 +167,7 @@ def wait_for_server_to_match(re, idx = nil, log: false)
       else
         true until (line = @server.gets || '').match?(re)
       end
-    rescue Errno::EBADF, Errno::ECONNREFUSED, Errno::ECONNRESET, IOError => e
+    rescue NoMethodError, Errno::EBADF, Errno::ECONNREFUSED, Errno::ECONNRESET, IOError => e
```

Comment:
I think we need the leave a comment in the code explaining why NoMethodError can happen here and why it is okay

Is it possible to address in some other way?